dashboard/src/actions/apps.ts

```diff
@@ -81,25 +84,36 @@ function getAppUpdateInfo(
   appVersion: string,
 ): ThunkAction<Promise<void>, IStoreState, null, AppsAction> {
   return async dispatch => {
-    const chartsInfo = await Chart.listWithFilters(chartName, currentVersion, appVersion);
-    let updateInfo: IChartUpdateInfo = {
-      repository: { name: "", url: "" },
-      latestVersion: "",
-    };
-    chartsInfo.forEach(c => {
-      const chartLatestVersion = c.relationships.latestChartVersion.data.version;
-      if (semver.gt(chartLatestVersion, currentVersion)) {
-        if (updateInfo.latestVersion && semver.gt(updateInfo.latestVersion, chartLatestVersion)) {
-          // The current update is newer than the chart version, do nothing
-        } else {
-          updateInfo = {
-            latestVersion: chartLatestVersion,
-            repository: c.attributes.repo,
-          };
-        }
+    dispatch(requestAppUpdateInfo());
+    try {
+      const chartsInfo = await Chart.listWithFilters(chartName, currentVersion, appVersion);
+      let updateInfo: IChartUpdateInfo = {
+        upToDate: true,
+        repository: { name: "", url: "" },
+        latestVersion: "",
+      };
+      if (chartsInfo.length > 0) {
+        // Initialize updateInfo with the first chart found
+        updateInfo = {
+          upToDate: true,
+          latestVersion: chartsInfo[0].relationships.latestChartVersion.data.version,
+          repository: chartsInfo[0].attributes.repo,
+        };
+        chartsInfo.forEach(c => {
```

Comment:
The first run through of this is not going to be useful since we already have `updateInfo` set to it, so it might make sense to drop the first element (e.g. with a for-loop `for (let i=1; i < chartsInfo.length; i++) {...}`).

That said, a cleaner way to do this might be to sort the array by the latest version. I haven't tried it, but something like this should work:

```
chartsInfo.sort((a, b) => semver.compare(a.relationships.latestChartVersion.data.version, b.relationships.latestChartVersion.data.version))
```

and then initialise updateInfo to the first element of the sorted array. 

Note that `semver.compare` above follows the conditions that the [`Array.sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Description) expects for the comparator function:

> compare(v1, v2): Return 0 if v1 == v2, or 1 if v1 is greater, or -1 if v2 is greater. Sorts in ascending order if passed to Array.sort().